### PR TITLE
Using GitOps and Grafana Operator with Amazon Managed Grafana

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*

Using GitOps and Grafana Operator with Amazon Managed Grafana

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Using GitOps and Grafana Operator with Amazon Managed Grafana
+
+## How to use this guide
+
+This Observability best practices guide is meant for developers and architects who want to understand how to use [grafana-operator](https://github.com/grafana-operator/grafana-operator#:~:text=The%20grafana%2Doperator%20is%20a,an%20easy%20and%20scalable%20way.) as a Kubernetes operator on your Amazon EKS cluster to create and manage the lifecycle of data sources such as Amazon Managed Service for Prometheus, Amazon CloudWatch, AWS X-Ray to Amazon Managed Grafana and create Grafana dashboards in Amazon Managed Grafana in a Kubernetes native way.
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:5]: Manage the lifecycle of "Amazon Managed Service for Prometheus, Amazon CloudWatch, AWS X-Ray". Is it? I thought this was only grafana resources? What does grafana-operator actually do in managing these other resources' lifecycles?

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Introduction to GitOps and Grafana Operator
+
+### What is GitOps and Flux
+
+GitOps is a software development and operations methodology that uses Git as the source of truth for deployment configurations. It involves keeping the desired state of an application or infrastructure in a Git repository and using Git-based workflows to manage and deploy changes. GitOps is a way of managing application and infrastructure deployment so that the whole system is described declaratively in a Git repository. It is an operational model that offers you the ability to manage the state of multiple Kubernetes clusters leveraging the best practices of version control, immutable artifacts, and automation. 
+
+Flux is a GitOps tool that automates the deployment of applications on Kubernetes. It works by continuously monitoring the state of a Git repository and applying any changes to a cluster. Flux integrates with various Git providers such as GitHub, [GitLab](https://dzone.com/articles/auto-deploy-spring-boot-app-using-gitlab-cicd), and Bitbucket. When changes are made to the repository, Flux automatically detects them and updates the cluster accordingly.
+
+### Pros of Flux
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:23]: Call this "Advantages of using Flux"

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* **Automated deployments**: Flux automates the deployment process, reducing manual errors and freeing up developers to focus on other tasks.
+* **Git-based workflow**: Flux leverages Git as a source of truth, which makes it easier to track and revert changes.
+* **Declarative configuration**: Flux uses [Kubernetes](https://dzone.com/articles/kubernetes-full-stack-example-with-kong-ingress-co) manifests to define the desired state of a cluster, making it easier to manage and track changes.
+
+### Cons of Flux
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:29]: Call this "Challenges in adopting Flux"

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:1]: This write up can be a bit more coherent. This needs to state the problem of managing resources in Grafana, then talk about ways to automate those. Then expand on using GitOps mechanism, and then get into doing this in K8s and explain why this is advantageous to do so. 
Finally call out the benefits of this solution while pointing users to the blog and workshop. 
The current one doesn't hit these points properly and is not going well with the principles of a "guide"

**Review by awsimaya — CHANGES_REQUESTED** [review]

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+One common challenge customers face while using Grafana is, in creating and managing the lifecycle of Grafana resources and Grafana dashboards in external Grafana instances such as Amazon Managed Grafana from their Kubernetes cluster. Customers face  challenges in finding ways to completely automate and manage infrastructure and application deployment of their whole system using Git based workflows which also includes creating of Grafana resources in Amazon Managed Grafana. In this Observability best practices guide, we will focus on the following topics:
+
+* Introduction on Grafana Operator - A Kubernetes operator to manage external Grafana instances from your Kubernetes cluster 
+* Introduction to GitOps - Automated mechanisms to create and manage your infrastructure using Git based workflows
+* Using Grafana Operator on Amazon EKS to manage Amazon Managed Grafana
+* Using GitOps with Flux on Amazon EKS to manage Amazon Managed Grafana
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:16]: "manage resources in Amazon Managed Grafana"

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+* **Steep learning curve**: Flux has a steep learning curve for new users and requires a deep understanding of Kubernetes and Git.
+
+## Using Grafana Operator on Amazon EKS to manage Amazon Managed Grafana
+
+As discussed in previous section, Grafana Operator enables us to use our Kubernetes cluster to create and manage the lifecyle of resources in Amazon Managed Grafana in a Kubernetes native way.  The below architecture diagram shows the demonstration of Kubernetes cluster as a control plane with using Grafana Operator to setup an identity with AMG, adding Amazon Managed Service for Prometheus as a data source and creating dashboards on Amazon Managed Grafana from Amazon EKS cluster in a Kubernetes native way.
+
+![GitOPS-WITH-AMG-2](../../../images/Operational/gitops-with-amg/gitops-with-amg-2.jpg)
+
+Please reference our post on [Using Open Source Grafana Operator on your Kubernetes cluster to manage Amazon Managed Grafana](https://aws.amazon.com/blogs/mt/using-open-source-grafana-operator-on-your-kubernetes-cluster-to-manage-amazon-managed-grafana/) for detailed demonstration of how to deploy the above solution on your Amazon EKS cluster.
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:49]: "Please refer to our post"

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+## Using GitOps with Flux on Amazon EKS to manage Amazon Managed Grafana
+
+As discussed above, Flux automates the deployment of applications on Kubernetes. It works by continuously monitoring the state of a Git repository such as GitHub and when changes are made to the repository, Flux automatically detects them and updates the cluster accordingly. Please reference the below architecture where we will be demonstrating how to use Grafana Operator from your Kubernetes cluster and GitOps mechanisms using Flux to add Amazon Managed Service for Prometheus as a data source and create dashboards in Amazon Managed Grafana in a Kubernetes native way. 
+
+![GitOPS-WITH-AMG-1](../../../images/Operational/gitops-with-amg/gitops-with-amg-1.jpg)
+
+Please reference the One Observability Workshop module - [GitOps with Amazon Managed Grafana](https://catalog.workshops.aws/observability/en-US/aws-managed-oss/gitops-with-amg). This module sets up required day 2 operational tooling such as the following on your EKS cluster:
```
Comment (awsimaya) [inline docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:57]: Say "Day 2" in quotes.

**Review by awsimaya — CHANGES_REQUESTED** [review]

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md (hunk 6 above)
Comment (elamaran11) [reply docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md]: Fixed.

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md (hunk 5 above)
Comment (elamaran11) [reply docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md]: Fixed.

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+To address this gap, AWS team collaborated with the [grafana-operator](https://github.com/grafana-operator/grafana-operator#:~:text=The%20grafana%2Doperator%20is%20a,an%20easy%20and%20scalable%20way.) team and submitted a [design proposal](https://github.com/grafana-operator/grafana-operator-experimental/pull/39)to support the integration of external Grafana instances. With this mechanism it will be possible to add external Prometheus compatible data sources (e.g., Amazon Managed Service for Prometheus) and create Grafana dashboards in external Grafana instances (e.g., Amazon Managed Grafana) from your Kubernetes cluster. This enables us to use our Kubernetes cluster to create and manage the lifecyle of resources in Amazon Managed Grafana in a Kubernetes native way. This ultimately enables us to use GitOps mechanisms using CNCF projects such as [Flux](https://fluxcd.io/) to create and manage the lifecyle of resources in Amazon Managed Grafana.
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md]: Fixed.

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md (hunk 4 above)
Comment (elamaran11) [reply docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:16]: Fixed.

File: docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+One common challenge customers face while using Grafana is, in creating and managing the lifecycle of Grafana resources and Grafana dashboards in external Grafana instances such as Amazon Managed Grafana from their Kubernetes cluster. Customers face  challenges in finding ways to completely automate and manage infrastructure and application deployment of their whole system using Git based workflows which also includes creating of Grafana resources in Amazon Managed Grafana. In this Observability best practices guide, we will focus on the following topics:
+
+* Introduction on Grafana Operator - A Kubernetes operator to manage external Grafana instances from your Kubernetes cluster 
+* Introduction to GitOps - Automated mechanisms to create and manage your infrastructure using Git based workflows
+* Using Grafana Operator on Amazon EKS to manage Amazon Managed Grafana
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/en/guides/operational/gitops-with-amg/gitops-with-amg.md:15]: Fixed.

**Comment by elamaran11** [comment]
@awsimaya Fixed all your feedback and also revamped few things here and there to align to your feedback. Please check and let me know.

**Pull request merged**